packages/blocks/src/note-block/keymap-controller.ts

```diff
@@ -0,0 +1,789 @@
+import type { BlockSelection, UIEventHandler } from '@blocksuite/block-std';
+import { assertExists, DisposableGroup } from '@blocksuite/global/utils';
+import type { BlockElement } from '@blocksuite/lit';
+import type { ReactiveController } from 'lit';
+import type { ReactiveControllerHost } from 'lit';
+
+import { moveBlockConfigs } from '../_common/configs/move-block.js';
+import { quickActionConfig } from '../_common/configs/quick-action/config.js';
+import { textConversionConfigs } from '../_common/configs/text-conversion.js';
+import { getBlockElementByModel } from '../_common/utils/index.js';
+import {
+  onModelElementUpdated,
+  updateBlockElementType,
+} from '../page-block/index.js';
+import {
+  changeTextSelectionSideways,
+  changeTextSelectionSidewaysToBlock,
+  changeTextSelectionToBlockStartEnd,
+  changeTextSelectionVertically,
+  ensureBlockInContainer,
+  getBlockSelectionBySide,
+  moveCursorToBlock,
+  moveCursorVertically,
+  pathToBlock,
+  selectBetween,
+  setBlockSelection,
+  setTextSelectionBySide,
+} from './utils.js';
+
+export class KeymapController implements ReactiveController {
```

Comment:
How do you distinguish the concept between "controller" and "service"? Is this also capable of being defined as a service?

Reply:
The controller here refers to the [reactive controller](https://lit.dev/docs/composition/controllers/) in lit.